data/campaigns/Winds_of_Fate/scenarios/09_Heart_Mountains.cfg

```diff
@@ -0,0 +1,639 @@
+        [filter_condition]
+            [lua]
+#ifdef EASY
+                code=<< return (wml.variables["turn_number"] % 6 == 0) >>
+#endif
+#ifdef NORMAL
+                code=<< return (wml.variables["turn_number"] % 3 == 0) >>
+#endif
+#ifdef HARD
+                code=<< return (wml.variables["turn_number"] % 1 == 0) >>
+#endif
+            [/lua]
+        [/filter_condition]
+        # Small and herbivorous creatures.
+        {SCATTER_UNITS 1 "Frost Stoat" 12 (
+            {EVERYWHERE}
+            terrain=A*
+        ) (side=2)}
+        {SCATTER_UNITS 1 "Elder Falcon" 12 (
+            {EVERYWHERE}
+        ) (side=2)}
+        {SCATTER_UNITS 1 "Woodland Boar" 4 (
+            {EVERYWHERE}
+            terrain=*^F*
+        ) (side=2)}
+        {SCATTER_UNITS 2 "Great Icemonax" 6 (
+            {EVERYWHERE}
+            terrain=Ha
+        ) (side=2)}
+        # Wolves
+        {SCATTER_UNITS 2 "Direwolf" 6 (
+            {EVERYWHERE}
+            terrain=*^F*
+        ) (side=3)}
+        # Bears
+        {SCATTER_UNITS 1 "Cave Bear" 12 (
+            {EVERYWHERE}
+            terrain=M*
+            [not]
+                terrain=Mv
+            [/not]
+        ) (side=4)}
+    [/event]
+
+    [event]
+        name=turn end
+        first_time_only=no
+        [filter_condition]
+            [lua]
+#ifdef EASY
+                code=<< return (wml.variables["turn_number"] % 12 == 0) >>
+#endif
+#ifdef NORMAL
+                code=<< return (wml.variables["turn_number"] % 6 == 0) >>
+#endif
+#ifdef HARD
+                code=<< return (wml.variables["turn_number"] % 2 == 0) >>
+#endif
+            [/lua]
+        [/filter_condition]
+        # Yeti
+        {SCATTER_UNITS 1 "Yeti" 12 (
+            {EVERYWHERE}
+            terrain=Ms
+        ) (side=5)}
+    [/event]
+
+    [event]
+        name=moveto
+        [filter]
+            x,y=5,11
+            side=1
+        [/filter]
+        [message]
+            speaker=narrator
+            message= _ "' BAK NEKST SPRING '"
+        [/message]
+    [/event]
+
+    [event]
+        name=moveto
+        [filter]
+            x,y=29,8
+            id=Gorlack
+        [/filter]
+        [delay]
+            time=1500
+        [/delay]
+        [fire_event]
+            name=dragon fight
+        [/fire_event]
+    [/event]
+
+    [event]
+        name=dragon fight
+        [unit]
+            side=7
+            id=Khrakrahs
+            name= _ "Khrakrahs"
+            type=Fire Dragon
+            profile=portraits/khrakrahs.png
+            x,y=38,1
+            facing=sw
+        [/unit]
+        [modify_side]
+            side=7
+            [ai]
+                [attacks]
+                    invalidate_on_gamestate_change=yes
+                    [filter_own]
+                        id=Khrakrahs
+                    [/filter_own]
+                    [filter_enemy]
+                        id=Gorlack
+                    [/filter_enemy]
+                [/attacks]
```

Comment:
This looks like valid syntax, according to the [documentation](https://wiki.wesnoth.org/Wesnoth_AI_Framework#The_.5Bai.5D_Tag_.E2.80.94_Aspects), so it might be a schema bug.

Reply:
The line appears necessary for proper functioning.
I've tested commenting out, removing, or setting invalidate_on_gamestate_change=no. In all those cases an assertion failure is triggered as soon as an enemy unit affected by that ai code finishes an attack in both S08 & 09:

Assertion failed!
File: src/units/map.hpp
Line 225
Expression: valid()

Reply:
That line definitely needs to be there, in order for the possible attacks in the attacks aspect to get updated after a gamestate change.  Otherwise the engine might try to do an attack that is not possible any more, which is probably the cause of the assert.

This sounds like schema needs to be updated.

Reply:
The schema almost certainly does need to be updated, I just need to figure out how to work that in there. The AI schema is extremely macro-heavy so my initial look at it left me more confused than enlightened.

However, if omitting this line causes an assert, then that's also a bug that needs to be fixed. WML should never be able to cause a crash.

Reply:
Yeah, true.  This seems like an inherent design flaw.  As I understand it, the attacks aspect is by default not updated on every gamestate change because that can take a lot of time, but then it might point to invalid attacks/units which in turn causes the assert.  Or something along those lines ...

In any case, that's outside the scope of this PR.  I was not able to reproduce the bug in a tests scenario, but did get the crash in WoF:S9 as described above.  I'll open a separate issue for it.

Reply:
…did I ever do this schema update? I can't remember…

Reply:
> Yeah, true. This seems like an inherent design flaw. As I understand it, the attacks aspect is by default not updated on every gamestate change because that can take a lot of time, but then it might point to invalid attacks/units which in turn causes the assert. Or something along those lines ...
> 
> In any case, that's outside the scope of this PR. I was not able to reproduce the bug in a test scenario, but did get the crash in WoF:S9 as described above. I'll open a separate issue for it.

Did you open this separate issue? If so, what's the number for it?